azure/Guidelines.md

```diff
@@ -1292,6 +1292,52 @@ In addition to distributed tracing, Azure also uses a set of common correlation
 |x-ms-client-request-id       |Both      |Optional. Caller-specified value identifying the request, in the form of a GUID with no decoration such as curly braces (e.g. `x-ms-client-request-id: 9C4D50EE-2D56-4CD3-8152-34347DC9F2B0`). If the caller provides this header the service **must** include this in their log entries to facilitate correlation of log entries for a single request. Because this header can be client-generated, it should not be assumed to be unique by the service implementation.
```

Comment:
To expand on this (and why it is important for services to not make assumptions about this value) - client request id is often used to group a set of "logical" client requests. Retries will use the same client request id. I believe that tools like the azure cli may use the same client request id for each request made in a single command execution.

Reply:
I agree with Johan here. I consider this header and it's value purely for client benefit. I don't think it has to be a guid. I think a service should specify a max length and perhaps valid characters. But I don't think more than this is required. Also, what is motivating that a service MUST log this value?